tests/test_state_value.py

```
@@ -0,0 +1,252 @@
+
+class TestStateValueInComponents:
+    """Test StateValue integration with typed Python components."""
+
+    def test_typed_component_with_scalar_output(self):
+        """Test that typed components can return scalar outputs via StateValue."""
+
+        class ScalarComponent(Component):
+            input_var = Input("Input", unit="W/m^2")
+            output_var = Output("Output", unit="K")
+
+            def solve(self, t_current, t_next, inputs):
```

Ruff (0.14.11)

Unused method argument: `t_current`

(ARG002)